### PR TITLE
AOCC: add v4.2.0

### PR DESCRIPTION
**Pull request by amd-toolchain-support**

### CODE REVIEW

File: var/spack/repos/builtin/packages/aocc/package.py [hunk 1]
```diff
@@ -56,7 +61,6 @@ class Aocc(Package):
     depends_on("zlib-api")
     depends_on("ncurses")
     depends_on("libtool")
-    depends_on("texinfo")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/aocc/package.py:59]: Why was this removed for every version?

Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/aocc/package.py:59]: This was a long-standing issue.  Texinfo is not actually required for any version currently in the recipe.

**Review by alalazo — APPROVED** [review]
